### DISCUSSION TITLE
Target/Cargill Hackathon Prep

### DISCUSSION THREAD

**Discussion by kneal** [go-vela/community#321] [General]
Hey Hackers!

**When:** Thursday, July 8, 2021 to Friday, July 9, 2021.

**Location:** _reach out to me for Zoom link if needed_

Welcome to the Vela Target/Cargill hackathon part two! Details will be added as they come to get more settled. 
 
##### Agenda for July 8th:

- Touchbase (5-10mins)
- Hack 
- Lunch break (12pm or up to teams)
- Hack
 
##### Agenda for July 9th:

- Touchbase (5-10mins)
- Hack 
- Lunch break @ 12pm or up to teams
- Hack
- Retro @ 3pm (30mins)
- HH

##### Example Hack Ideas:

- Enhancements for Templates
- Improved HTML rendering for logs in UI
- Build stats in the UI

**Comment by kneal** [comment]
For template enhancements, I was thinking we could look at maybe doing some templates with templates or thinking about some designs on how to make templates work with stages. 

I think both would be a pretty big chunk of work but I think the stages with templates one is the more tricky since existing templates aren't really designed super great to work with templating a stage. So we'd wanna like:

- Mini design session
- Proposal to double-check ourselves
- Write MVP

**Comment by kneal** [comment]
For build stats, I don't really have a ton of ideas on how they would fit into the overview page.

I can definitely see some value in success/failure for some teams and the length of time a build runs over time to see if a build is starting to take longer.

**Reply by wass3r** [reply]
I was imagining something more on a repo level (at least to start) unless that's what you meant by "overview"? Would be fairly trivial to add a "Stats" tab there to give it a home.

**Reply by kneal** [reply]
That makes sense, I've seen Circle do something similar but they call their page "Insights"

**Reply by jbrockopp** [reply]
We've (@Cappo @matt-fevold @jschulz79) started to work on this story.

We first had to augment some functionality in the Vela server to support querying builds by `branch` and `status`:

https://github.com/go-vela/server/pull/443

**Comment by JordanSussman** [comment]
Some additional ideas:

- Enable the `.vela.yml` to be templated without needing to utilize `templates`
- Enable builds to store artifacts within the Vela server
- Provide builds overview page

**Reply by kneal** [reply]
Do you have any ideas what the builds over page contain?

**Reply by wass3r** [reply]
Maybe a dashboard where you can create groupings of repos you are interested in monitoring and get a card (or something) for each repo in the group with recent build status, stats, etc. Not sure if that was the intent with that point.

**Reply by JordanSussman** [reply]
> Do you have any ideas what the builds over page contain?

Was thinking about something like https://github.com/go-vela/community/discussions/321#discussioncomment-975524  or https://github.com/go-vela/community/discussions/321#discussioncomment-976231.

**Comment by jbrockopp** [comment]
Another useful idea is enhancing the experience around orgs in Vela

Thinking that currently I can navigate to a repo via the UI in Vela like:

```
https://vela.company.com/<org>/<repo>
```

However, I can't navigate to a list of repos in the org like:

```
https://vela.company.com/<org>
```

While I'm specifically calling out functionality in the UI, this functionality would probably involve a couple different repos.

**Reply by JordanSussman** [reply]
@JayCeeJr was tentatively planning on working on this once https://github.com/go-vela/ui/pull/414 is done.

**Comment by jbrockopp** [comment]
Another idea may be to improve the experience around secrets and navigation in Vela

As an example, there isn't a great way to navigate to `shared` secrets in the UI w/o manually updating the URL path:

```
https://vela.company.com/-/secrets/native/shared/<org>/<team>
```

A somewhat related but different example is trying to navigate to `org` secrets in the UI.

Again, you can manually update the URL path:

```
https://vela.company.com/-/secrets/native/org/<org>
```

Or you have to navigate to that page via the `repo` secrets page and click on the `Manage Org Secrets` button:

```
https://vela.company.com/-/secrets/native/repo/<org>/<repo>
```

While I'm specifically calling out functionality in the UI, this functionality would probably involve a couple different repos.

**Reply by wass3r** [reply]
This might necessitate https://github.com/go-vela/community/discussions/321#discussioncomment-975524, though we can probably hack something in.
